neuraxle/steps/util.py

```diff
@@ -517,7 +511,45 @@ def get_cache_path_for(self, data_input) -> str:
         raise NotImplementedError()
 
 
-class PickleValueCachingWrapper(BaseValueCachingWrapper):
+class OutputTransformerMixin:
+    """
+    Base output transformer step that can modify data inputs, and expected_outputs at the same time.
+    """
+
+    def handle_transform(self, data_container: DataContainer) -> DataContainer:
+        """
+        Handle transform by updating the data inputs, and expected outputs inside the data container.
+
+        :param data_container:
+        :return:
+        """
+        di_eo = (data_container.data_inputs, data_container.expected_outputs)
+        new_data_inputs, new_expected_outputs = self.transform(di_eo)
+
+        data_container.set_data_inputs(new_data_inputs)
+        data_container.set_expected_outputs(new_expected_outputs)
+
+        current_ids = self.hash(data_container.current_ids, self.hyperparams, data_container.data_inputs)
+        data_container.set_current_ids(current_ids)
+
+        return data_container
+
+    def handle_fit_transform(self, data_container: DataContainer) -> ('BaseStep', DataContainer):
+        """
+        Handle transform by fitting the step,
+        and updating the data inputs, and expected outputs inside the data container.
+
+        :param data_container:
+        :return:
+        """
+        new_self = self.fit(data_container.data_inputs, data_container.expected_outputs)
```

Comment:
Does the fit needs a `di_eo` tuple too? I feel like it would need that too, although I didn't think more why.

Reply:
Like: 
`        new_self = self.fit((data_container.data_inputs, data_container.expected_outputs), None)`

Reply:
outdated